src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3241,18 +3242,37 @@ static size_t large_page_init_decide_size() {
     return 0;
   }
 
-#if defined(IA32) || defined(AMD64)
-  if (size > 4*M || LargePageSizeInBytes > 4*M) {
-    WARN("JVM cannot use large pages bigger than 4mb.");
-    return 0;
-  }
+  // Check if OS version is 11 or greater
+  if (win32::is_windows_11_or_greater() || win32::is_windows_server_2022_or_greater()) {
```

Comment:
Do we want or need an empty if clause here? Why don't we just check to make sure we're *not* Windows 11 or greater and *not* Windows Server 2022 or greater and, if so, execute the existing logic?

Reply:
This was to align with the existing flow @macarte Please comment

Reply:
changed